File: argentum-api/src/main/java/net/devium/argentum/rest/model/request/StatusRequest.java
package net.devium.argentum.rest.model.request;

import net.devium.argentum.jpa.StatusEntity;

public class StatusRequest {
    private Long id;
    private String internalName;
    private String displayName;
    private String color;

    public Long getId() {
        return id;
    }

    public String getInternalName() {
        return internalName;
    }

    public String getDisplayName() {
        return displayName;
    }

    public String getColor() {
        return color;
    }

    public StatusEntity toEntity() {
        return new StatusEntity(
                id != null ? id : -1,
                internalName != null ? internalName : "",
                displayName != null ? displayName : "",
                color != null ? color : "#ffffff"
        );
    }
}

File: argentum-ui/src/app/common/rest-service/request/status-request.ts
import { Status } from '../../model/status';
export class StatusRequest {
  id: number;
  internalName: string;
  displayName: string;
  color: string;
}

export function fromStatus(status: Status): StatusRequest {
  return {
    id: status.id,
    internalName: status.internalName,
    displayName: status.displayName,
    color: status.color
  };
}
